### DISCUSSION TITLE
auto apply

### DISCUSSION THREAD

**Discussion by timsofteng** [twpayne/chezmoi#2269] [General]
Hello. It's annoying to apply chezmoi after every editing.
Is it possible to to do automatic apply on files saving?

**Comment by twpayne** [comment]
You can use `chezmoi edit --apply $FILENAME` or [configure your editor to run chezmoi apply when you save](https://www.chezmoi.io/user-guide/tools/editor/).

**Comment by timsofteng** [comment]
I used to use vim for dotfiles and just do `vim somedotfile`
To be honest I do not use special command to edit dotfiles. Is it possible to do it with chezmoi only? Like watcher or hook?

**Reply by twpayne** [reply]
#2279 (just merged) adds instructions for [using chezmoi with Watchman](https://github.com/twpayne/chezmoi/commit/e2ec7c5308f589b8b4a3e9f0607d2d218dbb2c98) and adds a `--watch` command to `chezmoi edit`.

As an alternative to a hook you can use an alias, for example:

```bash
alias cze='chezmoi edit --apply'
```

and then you can do

```console
$ cze .bashrc
```

and the changes will be applied when you quit your editor.

**Reply by twpayne** [reply]
@timsofteng hopefully this answers your question.
